src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
there is a new service that is supposed to help make every services in azure able to "react" to events. similar to how Azure Functions hooks on lots of things